### ISSUE TITLE
General Question SQL methods compatibility 

### ISSUE THREAD

**Issue by waltergrimmvie** [praeclarum/sqlite-net#783]
I am aware that this is a complicated topic and all people are putting tons of effort into it,  I landed here from official MS site.

I did a SQLite in WinForms for offline storage of specific data, I was able to be finished in one day, with standard SQL knowledge select, insert, update, delete.

This here looks very cryptical to me. Is there no way to use the standard SQL commands here, or do I really need to dig into this here, or is there any plan on the horizon to add some functionality to achieve near winforms methods.

thanxs for the info, 

Walter

**Comment by ericsink** [comment]
sqlite-net is a somewhat high-level wrapper around SQLite, providing light ORM-ish features.

It's not clear what you are looking for, but your note has two clues I can use to create guesses:

1.  "standard SQL commands" suggests you want something lower-level than sqlite-net.  If so, you could just use SQLitePCL.raw (which I maintain, and upon which sqlite-net is built), which gives an API very similar to the native C API provided by SQLite itself.

2.  "winforms methods" suggests maybe you want something ADO-ish.  If so, you could use Microsoft.Data.Sqlite (which, FWIW, is also built on SQLitePCL.raw).

If neither of these guesses is helpful, please give more detail about what coding in "SQLite in WinForms" was like.

**Comment by waltergrimmvie** [comment]
Hi, thank for the great info, I will check these, I have not found these yet, since I followed the official MS Xamarin site, if you like I could send you my one page for sqlite in winforms FYI, it's really very,very easy to understand for starters

Thanxs

**Comment by ericsink** [comment]
Yeah, just post the code (or some snippet showing something to do with sqlite) here.  That'll probably make it easy to figure out the best path for you.

**Comment by waltergrimmvie** [comment]
Thanks very much, MS xamarin links to you, without mentioning the ms.data.sqlite anywhere, I would be glad to check alternatives, why are you running this approach parallel to MS, might be Interesting for other people looking around

**Comment by ericsink** [comment]
sqlite-net has been around for much longer than Microsoft.Data.Sqlite, so it has built up a following.

People who have experience with ADO.NET on the desktop may find Microsoft.Data.Sqlite easier.

People who are starting from scratch may find sqlite-net easier.

There are quite a few ways to access SQLite from .NET code.

**Comment by waltergrimmvie** [comment]
Hi, I have done some research, and all over they talk about mono, which is outdated, and that Microsoft.Data.Sqlite is for win forms,  are you sure this works for xamarin cross platform

**Comment by waltergrimmvie** [comment]
OK,  just about everything I get on Google tells me that this does not work on xamarin, since you are developing a sqlite package you surely now this very well, so I will give it a try, but even the gitrep has this on an asp.net site
I will Inform you of my findings

**Comment by ericsink** [comment]
The repo for Microsoft.Data.Sqlite is:

https://github.com/aspnet/EntityFrameworkCore

(it's part of Entity Framework Core's repo)

If you want more authoritative answers to questions about Xamarin compatibility, you could ask there.
